src/test/test_ptr_slow.c

```diff
@@ -0,0 +1,86 @@
+/* Copyright (c) 2001-2004, Roger Dingledine.
+ * Copyright (c) 2004-2006, Roger Dingledine, Nick Mathewson.
+ * Copyright (c) 2007-2019, The Tor Project, Inc. */
+/* See LICENSE for licensing information */
+
+#include "orconfig.h"
+#include "core/or/or.h"
+#include "test/test.h"
+
+#include <stdint.h>
+#include <limits.h>
+
+/** Assert that <b>a</b> can be cast to void * and back. */
+static void
+assert_int_voidptr_roundtrip(int a)
+{
+  intptr_t ap = (intptr_t)a;
+  void *b = (void *)ap;
+  intptr_t c = (intptr_t)b;
+  void *d = (void *)c;
+
+  tt_assert(ap == c);
+  tt_assert(b == d);
+
+ done:
+  return;
+}
+
+static void
+test_int_voidstar_interop(void *arg)
+{
+  int a;
+  (void)arg;
+
```

Comment:
It's important that we check negative integers.

Let's check -1024…0 and INT_MIN…(INT_MIN+1024)?

Reply:
Did so in 280109473fe30d1e67f8db09ce279cc1f714a682.